engine/core/source/org/pentaho/reporting/engine/classic/core/layout/output/AbstractReportProcessor.java

```diff
@@ -1460,7 +1460,7 @@ public void processReport() throws ReportProcessingException {
             + ": Report processing failed.", e );
         throw new ReportProcessingException( "Failed to process the report", e );
       }
-      fireProcessingFinished( new ReportProgressEvent( this, getPhysicalPageCount() ) );
+      fireProcessingFinished( new ReportProgressEvent( this, getPhysicalPageCount(), getLogicalPageCount() ) );
       if ( AbstractReportProcessor.logger.isDebugEnabled() ) {
```

Comment:
Your change uncovered a bug. The "getPhysicalPageCount()" call is wrong. It should be the logical-page we are querying here (to keep it in common with all the other page-event uses of the page-count). 

The reporting engine can have reports where multiple physical pages form a single logical page (page-spanning or poster-print mode). 

Whenever we query the total pages (@see calculatePageCount(..)) we actually count the logical pages (or in other words: How many page-breaks did we detect.) Thus for the sake of consistency, change this "getPhysicalPageCount" call into a getLogicalPageCount() call. Then our process-state never reports a page-number that is larger than the total page number.